### DISCUSSION TITLE
How can I disable babel detection/auto load?

### DISCUSSION THREAD

**Discussion by Grohden** [storybookjs/storybook#28530] [Help] [answered]
### Summary

We're using a monorepo like this

```
- babel.config.js
- web/
  - nextjs + storybook for web
```

I'm trying to make SB use swc like next, but due to babel detection (?) walking up it finds the babel config and uses babel instead of swc... if I remove the top level babel it uses swc, but I can't do that

So its possible to make sb prefer swc first rather than babel? the next config doesn't need `.swcrc` so I'm not using one

### Additional information

```js
const config = {
  ....,
  framework: {
    name: '@storybook/nextjs',
    options: {
      // storybook doesn't support {(phase: string) => import('next').NextConfig} as a next.config.js 
      // so we have to hack around making a file that just calls it with no buildphase to make it work
      nextConfigPath: resolve(__dirname, '../next.config.storybook.js'),
    },
  },
  // ....
  addons: [
    // 'storybook-addon-swc', // also works as long as I don't have a babelrc file
    '@storybook/addon-essentials',
    '@storybook/addon-a11y',
    'storybook-addon-apollo-client',
    '@chromatic-com/storybook',
    '@storybook/addon-webpack5-compiler-swc',
  ],
}
```

**Comment by valentinpalkovic** [comment, marked as answer]
If you want to use SWC despite the presence of a .babelrc file you can force it in your next.config.js file.

https://nextjs.org/docs/messages/swc-disabled#possible-ways-to-fix-it

Storybook will respect this setting and will also use SWC

**Reply by Grohden** [reply]
that works, thanks! I thought that it was an issue with SB cuz next was already not picking up the upper babel file

**Reply by valentinpalkovic** [reply]
So it might be the case that Next.js only looks in the current working directory for a babelrc file, whereas Storybook traverses the file tree until it finds one.
